rust/reflection/src/safe_buffer.rs

```diff
@@ -0,0 +1,392 @@
+            // SAFETY: the buffer was verified during construction.
+            unsafe { get_field_vector(table, &field) }
+        } else {
+            Err(FlatbufferError::FieldNotFound())
+        }
+    }
+
+    /// Gets a Table table field given its exact type. Returns [None] if the field is not set. Returns error if
+    /// the [table] doesn't match the buffer or
+    /// the [field_name] doesn't match the [table] or
+    /// the field type doesn't match.
+    pub fn get_field_table(
+        &self,
+        table: &'a Table,
+        field_name: &str,
+    ) -> FlatbufferResult<Option<Table<'a>>> {
+        if !self.contains_table(table)? {
+            return Err(FlatbufferError::InvalidTableOrStruct());
+        }
+
+        if let Some(field) = self.find_field_by_name(table.loc(), field_name)? {
+            // SAFETY: the buffer was verified during construction.
+            unsafe { get_field_table(table, &field) }
+        } else {
+            Err(FlatbufferError::FieldNotFound())
+        }
+    }
+
+    /// Returns the value of any table field as a 64-bit int, regardless of what type it is. Returns default integer if the field is not set or error if
+    /// the value cannot be parsed as integer or
+    /// the [table] doesn't match the buffer or
+    /// the [field_name] doesn't match the [table].
+    /// [num_traits](https://docs.rs/num-traits/latest/num_traits/cast/trait.NumCast.html) is used for number casting.
+    pub fn get_any_field_integer(&self, table: &Table, field_name: &str) -> FlatbufferResult<i64> {
+        if !self.contains_table(table)? {
+            return Err(FlatbufferError::InvalidTableOrStruct());
+        }
+
+        if let Some(field) = self.find_field_by_name(table.loc(), field_name)? {
+            // SAFETY: the buffer was verified during construction.
+            unsafe { get_any_field_integer(table, &field) }
+        } else {
+            Err(FlatbufferError::FieldNotFound())
+        }
+    }
+
+    /// Returns the value of any table field as a 64-bit floating point, regardless of what type it is. Returns default float if the field is not set or error if
+    /// the value cannot be parsed as float or
+    /// the [table] doesn't match the buffer or
+    /// the [field_name] doesn't match the [table].
+    pub fn get_any_field_float(&self, table: &Table, field_name: &str) -> FlatbufferResult<f64> {
+        if !self.contains_table(table)? {
+            return Err(FlatbufferError::InvalidTableOrStruct());
+        }
+
+        if let Some(field) = self.find_field_by_name(table.loc(), field_name)? {
+            // SAFETY: the buffer was verified during construction.
+            unsafe { get_any_field_float(table, &field) }
+        } else {
+            Err(FlatbufferError::FieldNotFound())
+        }
+    }
+
+    /// Returns the value of any table field as a string, regardless of what type it is. Returns empty string if the field is not set. Returns error if
```

Comment:
Nit: if this is public API, it would be nice to point out that the value gets stringified and not reinterpreted as a string, so for example an int (`1234`) becomes `"1234"` and not `"\x01\x02\x03\x04"` (+/- endianness)